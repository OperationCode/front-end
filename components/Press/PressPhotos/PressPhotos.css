.photos {
  padding-top: 15px;
  display: flex;
  flex-flow: row wrap;
  justify-content: space-around;
  align-items: center;
}

.photos > img {
  max-width: 350px;
  width: auto;
  padding: 0.75rem;
}

@media screen and (max-width: 499px) {
  .photos > img {
    width: 100%;
  }
}
