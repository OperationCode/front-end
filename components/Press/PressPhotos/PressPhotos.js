import React from 'react';
import { s3 } from 'common/constants/urls';
import styles from './PressPhotos.module.css';

function PressPhotos() {
  return (
    <div className={styles.photos}>
      <img
        src={`${s3}photo_code-conference-la.jpg`}
        alt="David Molina speaks to Code Conference LA 2016 attendees."
      />
      <img
        src={`${s3}photo_red-hat-summit-1.jpg`}
        alt="Operation Code members pose at Red Hat Summit 2017"
      />
      <img
        src={`${s3}photo_node-summit.jpg`}
        alt="Conrad Hollomon presents in front of the Node Summit 2016 audience."
      />
      <img
        src={`${s3}photo_angelhack-boston.jpg`}
        alt="Operation Code developers pose at AngelHack Boston 2017."
      />
      <img
        src={`${s3}photo_utah-meetup.jpg`}
        alt="Ken Collier leads a discussion at the Utah Operation Code meetup."
      />
    </div>
  );
}

export default PressPhotos;
