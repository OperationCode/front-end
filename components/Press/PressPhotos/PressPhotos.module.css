.photos {
  padding-top: 15px;
  display: flex;
  flex-flow: row wrap;
  justify-content: space-around;
  align-items: center;
}

.photos > img {
  max-width: 350px;
  width: auto;
  padding: 0.75rem;
}

@media screen and (--small-viewport) {
  .photos > img {
    width: 100%;
  }
}
