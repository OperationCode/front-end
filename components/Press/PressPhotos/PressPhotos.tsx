import { s3 } from 'common/constants/urls';
import Image from 'next/image';
import { cx } from 'common/utils/cva';

const pressPhotos = [
  {
    src: `${s3}photo_code-conference-la.jpg`,
    alt: 'David Molina speaks to Code Conference LA 2016 attendees.',
  },
  {
    src: `${s3}photo_red-hat-summit-1.jpg`,
    alt: 'Operation Code members pose at Red Hat Summit 2017',
  },
  {
    src: `${s3}photo_node-summit.jpg`,
    alt: 'Conrad Hollomon presents in front of the Node Summit 2016 audience',
  },
  {
    src: `${s3}photo_angelhack-boston.jpg`,
    alt: 'Operation Code developers pose at AngelHack Boston 2017.',
    className: 'object-top',
  },
  {
    src: `${s3}photo_utah-meetup.jpg`,
    alt: 'Ken Collier leads a discussion at the Utah Operation Code meetup.',
  },
];

function PressPhotos() {
  return (
    <div className="pt-4 flex flex-wrap mx-auto items-center justify-center gap-4">
      {pressPhotos.map(photo => (
        <div key={photo.src} className="relative aspect-video h-80">
          <Image
            src={photo.src}
            alt={photo.alt}
            layout="fill"
            className={cx('object-cover', photo.className)}
          />
        </div>
      ))}
    </div>
  );
}

export default PressPhotos;
