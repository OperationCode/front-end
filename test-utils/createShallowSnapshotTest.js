import ShallowRenderer from 'react-test-renderer/shallow';

/**
 * Used to create a snapshot test off of surface leaves of the tree.
 * Useful for regression tests on containers, components utilizing something that requires large
 * mocking, or very large components.
 *
 * @export
 * @param {*} Component
 */
export default Component => {
  const shallowRenderer = new ShallowRenderer();
  const tree = shallowRenderer.render(Component);

  expect(tree).toMatchSnapshot();
};
