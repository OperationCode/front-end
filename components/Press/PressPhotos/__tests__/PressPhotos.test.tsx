import createShallowSnapshotTest from 'test-utils/createShallowSnapshotTest';

import PressPhotos from '../PressPhotos';

describe('PressPhotos', () => {
  it('should render with no props passed', () => createShallowSnapshotTest(<PressPhotos />));
});
