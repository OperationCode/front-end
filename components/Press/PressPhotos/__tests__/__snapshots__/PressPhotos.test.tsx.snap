// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PressPhotos > should render with no props passed 1`] = `
<div
  className="pt-4 flex flex-wrap mx-auto items-center justify-center gap-4"
>
  <div
    className="relative aspect-video h-80"
  >
    <MockedNextImage
      alt="David Molina speaks to Code Conference LA 2016 attendees."
      className="object-cover"
      layout="fill"
      src="https://operation-code-assets.s3.us-east-2.amazonaws.com/photo_code-conference-la.jpg"
    />
  </div>
  <div
    className="relative aspect-video h-80"
  >
    <MockedNextImage
      alt="Operation Code members pose at Red Hat Summit 2017"
      className="object-cover"
      layout="fill"
      src="https://operation-code-assets.s3.us-east-2.amazonaws.com/photo_red-hat-summit-1.jpg"
    />
  </div>
  <div
    className="relative aspect-video h-80"
  >
    <MockedNextImage
      alt="Conrad Hollomon presents in front of the Node Summit 2016 audience"
      className="object-cover"
      layout="fill"
      src="https://operation-code-assets.s3.us-east-2.amazonaws.com/photo_node-summit.jpg"
    />
  </div>
  <div
    className="relative aspect-video h-80"
  >
    <MockedNextImage
      alt="Operation Code developers pose at AngelHack Boston 2017."
      className="object-cover object-top"
      layout="fill"
      src="https://operation-code-assets.s3.us-east-2.amazonaws.com/photo_angelhack-boston.jpg"
    />
  </div>
  <div
    className="relative aspect-video h-80"
  >
    <MockedNextImage
      alt="Ken Collier leads a discussion at the Utah Operation Code meetup."
      className="object-cover"
      layout="fill"
      src="https://operation-code-assets.s3.us-east-2.amazonaws.com/photo_utah-meetup.jpg"
    />
  </div>
</div>
`;
